### PR TITLE
[ENH] refactor _threads_to_use to _n_jobs

### LINKED ISSUE #102: [ENH] refactor BaseClassifier attribute _threads_to_use to _n_jobs
**Is your feature request related to a problem? Please describe.**

way back when, the internal copy of number of jobs in the BaseClassifier was set to be _threads_to_use (I think this was my bad). It should be _n_jobs to be consistent with other notation. Refactoring touches a lot of files, but also highlights that many classifiers are using their child class n_jobs  variable rather than the parent class _n_jobs. Thus it becomes more disruptive and dangerous to refactor,. See here
https://github.com/scikit-time/scikit-time/pull/101
I have closed the above PR while other large changes go on and just raised the issue here to be solved once the code base settles down.

### PR DESCRIPTION
**Pull request by TonyBagnall**
#### Reference Issues/PRs

fixes #102 

#### What does this implement/fix? Explain your changes.

Simple refactor of _n_threads_to_use to _n_jobs

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -12,14 +12,6 @@ addopts =
...
     --ignore docs
     --doctest-modules
-    --durations 10
-    --timeout 600
-    --cov aeon
-    --cov-report xml
-    --cov-report html
-    --showlocals
-    --matrixdesign True
-    -n auto
```
Comment (MatthewMiddlehurst) [inline setup.cfg:22]: The classic

**Review by MatthewMiddlehurst — CHANGES_REQUESTED** [review]

**Comment by patrickzib** [comment]
Hey @TonyBagnall , thanks a lot for touching these again! It is a monster.

I am afraid we are not there yet. There are quite a few classes using `self.n_jobs`. 

For the base classes, I do not even understand where the variable `self.n_jobs` was set (possibly being a bug?):

- Clustering: https://github.com/aeon-toolkit/aeon/blob/e7e4592f6135f4fe3c5cb220a60541002719ec5f/aeon/clustering/base.py#L71

- Classification:  https://github.com/aeon-toolkit/aeon/blob/e7e4592f6135f4fe3c5cb220a60541002719ec5f/aeon/regression/base.py#L153

- Regression: https://github.com/aeon-toolkit/aeon/blob/e7e4592f6135f4fe3c5cb220a60541002719ec5f/aeon/regression/base.py#L153


IMO: it would be best to refactor all occurrences of `self.n_jobs` to `self._n_jobs`, too, and there are > 100 occurrences of self.n_jobs :/

**Comment by patrickzib** [comment]
Sorry, if I am missing something. But how is `self.n_jobs` initialized in these base classes?

File: setup.cfg (hunk 1 above)
Comment (TonyBagnall) [reply setup.cfg:22]: every. bloody. time. :)

**Comment by patrickzib** [comment]
If we agree on something that is meaningful, I can pick it up and do the changes to the estimators.

Could you explain the logic behind self.n_jobs for me? It seems like a FK twist.

Right now, it seems like every estimator has to set `self.n_jobs` from the same parameter passed to the class - next, each base-class checks  `self.n_jobs` and sets it to `self._n_jobs`. Yet, most classifiers still use `self.n_jobs`, as they are not aware, that the base class set `self._n_jobs`.

**Comment by MatthewMiddlehurst** [comment]
This was actually me. I put it in the base class to avoid coping the check method in every estimator. I feel having a method will just run into the same issue of people using the estimator `self.n_jobs` over it.

Can either make people process `self.n_jobs` in their own estimator, or just document better/mention it in reviews.

**Comment by TonyBagnall** [comment]
> 

I dont feel too strongly about it, but I think it should probably all be dealt with in the base class and also be a base class parameter. That or take it out of base class, its neither one nor the other at the moment.

**Comment by TonyBagnall** [comment]
> 

if we defaulted it to None in the base class it would then fail tests if people reset it in their subclasses?

**Comment by TonyBagnall** [comment]
what do you think @patrickzib ? Do this PR now and sort out later or close this PR and fix all together? I'm fine either way, I just dont like PRs having around :)

**Comment by MatthewMiddlehurst** [comment]
I think we should merge this and create a separate issue to convert classifiers/regressors which use `self.n_jobs`. I think the new name is better.

It's difficult to enforce the usage of `check_n_jobs`/`self._n_jobs` other than by review and documentation IMO. I think the current base class way works fine in reducing the amount of copied boilerplate, it just has a bad name.

**Comment by MatthewMiddlehurst** [comment]
Can go through each estimator and change it in this PR also, I don't mind. Not sure how much work/extra file changes that would be though.

**Review by MatthewMiddlehurst — APPROVED** [review]

**Pull request merged**